pipelines/rj_crm__get_history_data/tasks.py

```
@@ -313,23 +218,144 @@ def build_historico_dataframe(results: List[Dict[str, Any]]) -> None:
+    insert_values = ", ".join(f"S.{col}" for col in all_columns)
+
+    query = f"""
+        MERGE {target} AS T
+        USING {source} AS S
+        ON {on_clause}
+        WHEN MATCHED THEN
+          UPDATE SET {update_clause}
+        WHEN NOT MATCHED THEN
+          INSERT ({insert_columns})
+          VALUES ({insert_values})
+    """
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)